Formula/libirecovery.rb

```diff
@@ -0,0 +1,27 @@
+class Libirecovery < Formula
+  desc "Library and utility to talk to iBoot/iBSS via USB"
+  homepage "https://www.libimobiledevice.org/"
+  url "https://github.com/libimobiledevice/libirecovery/releases/download/1.0.0/libirecovery-1.0.0.tar.bz2"
+  sha256 "cda0aba10a5b6fc2e1d83946b009e3e64d0be36912a986e35ad6d34b504ad9b4"
+  license "LGPL-2.1-only"
+
+  head do
+    url "https://git.libimobiledevice.org/libirecovery.git"
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+  end
+
+  def install
+    system "./autogen.sh" if build.head?
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--enable-debug-code"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/irecovery", "--help"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
libirecovery requires an iOS to be connected; running the `irecovery` utility with an argument that isn't `--help` and `--version` will just have it error out.

Reply:
As long as it exercises some of the mainline functionality, even a failure can be an acceptable test. For example, see https://github.com/Homebrew/homebrew-core/blob/master/Formula/shellcheck.rb#L37

Reply:
> If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

This also goes for devices being connected to the machine.